### PR TITLE
Release notes for M13

### PR DESCRIPTION
**Pull request by kchalkias**

### CODE REVIEW

File: docs/source/release-notes.rst [hunk 1]
```diff
@@ -4,11 +4,51 @@ Release notes
...
+Before M13, we tried to serialise everything reachable from the stack on each flow checkpoint. As flows are arbitrary
+code in which it is convenient to do many things, we ended up pulling in a lot of objects that didn't make sense
+to put in a checkpoint, such as ``Thread`` and ``Connection``. To minimize serialization cost and increase
+security by not allowing certain classes to be serialized, we now support class blacklisting
+that will return an ``IllegalStateException`` if such a class is encountered during a checkpoint. Blacklisting supports
+superclass and superinterface inheritance and always precedes ``@CordaSerializable`` annotation checking.
+
+We've also started working on improving user experience when searching, by adding a new RPC to support fuzzy matching
+of X.500 names. Our aim is to extend our work in usability and we are planning to apply AI and ML techniques in the
+near future.
```
Comment (josecoll) [inline docs/source/release-notes.rst]: AI and ML techniques ??? Is this for real?

Comment (kchalkias) [inline docs/source/release-notes.rst]: I made a reference on using Jaro–Winkler distance for string similarity and we will probably utilise Apache Lucene in the near future. In some sense these are the first steps for AI support

File: docs/source/release-notes.rst [hunk 2]
```diff
@@ -4,11 +4,51 @@ Release notes
...
+
+Milestone 13
+------------
 
-Certificate checks have been enabled for much of the identity service, with additional checks coming targetted at M13.
-These are part of the confidential identities work, and ensure that parties are actually who they claim to be by checking
-their certificate path back to the network trust root (certificate authority).
+Following our first public beta in M12, this release continues the work on API stability and user friendliness. Apart
+from bug fixes and code refactoring, there are also significant improvements mainly in the Vault Query and the
+Identity Service, respectively (for more detailed information about what has changed, see :doc:`changelog`).
```
Comment (gendal) [inline docs/source/release-notes.rst]: lose "mainly" and "respectively"

File: docs/source/release-notes.rst [hunk 3]
```diff
@@ -4,11 +4,51 @@ Release notes
...
+More specifically:
+
+The long awaited new **Vault Query** service makes its debut in this release and provides advanced vault query
+capabilities using criteria specifications (see ``QueryCriteria``), sorting, and pagination. Criteria specifications
+enable selective filtering with and/or composition using multiple operator primitives on standard attributes stored in
+Corda internal vault tables (eg. vault_states, vault_fungible_states, vault_linear_states), and also on custom contract
+state schemas defined by CorDapp developers when modelling new contract types. Custom queries are specifiable using a
+simple but sophisticated builder DSL (see ``QueryCriteriaUtils``). The new Vault Query service is usable by flows and by
+RPC clients alike via two simple API functions: ``queryBy()`` and ``trackBy()``. The former, provides point-in-time
```
Comment (gendal) [inline docs/source/release-notes.rst]: lose the comma after former

File: docs/source/release-notes.rst [hunk 4]
```diff
@@ -4,11 +4,51 @@ Release notes
...
+
+Before M13, we tried to serialise everything reachable from the stack on each flow checkpoint. As flows are arbitrary
+code in which it is convenient to do many things, we ended up pulling in a lot of objects that didn't make sense
+to put in a checkpoint, such as ``Thread`` and ``Connection``. To minimize serialization cost and increase
+security by not allowing certain classes to be serialized, we now support class blacklisting
+that will return an ``IllegalStateException`` if such a class is encountered during a checkpoint. Blacklisting supports
+superclass and superinterface inheritance and always precedes ``@CordaSerializable`` annotation checking.
+
+We've also started working on improving user experience when searching, by adding a new RPC to support fuzzy matching
+of X.500 names. Our aim is to extend our work in usability and we are planning to apply AI and ML techniques in the
```
Comment (gendal) [inline docs/source/release-notes.rst]: i'd prefer us not to make these sorts of commitments... let's shout about them if/when they're done :)

File: docs/source/release-notes.rst [hunk 5]
```diff
@@ -4,11 +4,51 @@ Release notes
...
+identities work, and ensure that parties are actually who they claim to be by checking their certificate path back to
+the network trust root (certificate authority).
+
+To deal with anonymized keys, we've also implemented a deterministic key derivation function that combines logic
+from the HMAC-based Extract-and-Expand Key Derivation Function (HKDF) protocol and the BIP32 hardened
+parent-private-key -> child-private-key scheme. This function currently supports the following algorithms:
+ECDSA secp256K1, ECDSA secpR1 (NIST P-256) and EdDSA ed25519. We are now very close to fully supporting anonymous
+identities so as to increase privacy even against validating notaries.
+
+Before M13, we tried to serialise everything reachable from the stack on each flow checkpoint. As flows are arbitrary
```
Comment (gendal) [inline docs/source/release-notes.rst]: "we tried to..." makes it sounds as if it was a conscious desire.  How about: "We have further tightened the set of objects which Corda will attempt to serialise from the stack during flow checkpointing"?

**Comment by kchalkias** [comment]
thanks @gendal, all issues addressed.

**Pull request merged**